### PR TITLE
Firefox fixes and improvements

### PR DESCRIPTION
**Pull request by dzedro**
- Related ticket: https://progress.opensuse.org/issues/41405
- Needles: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/955
- Verification run:
[SLED-15](http://10.100.12.155/tests/7890)
[SLED-12SP4](http://10.100.12.155/tests/7898)
[SLED-12SP3](http://10.100.12.155/tests/7897)
[SLES-15](http://10.100.12.155/tests/7891)
[SLES-12SP4](http://10.100.12.155/tests/7892)
[SLES-12SP3](http://10.100.12.155/tests/7894)
[SLES-12SP2](http://10.100.12.155/tests/7893)
[SLES-12SP1](http://10.100.12.155/tests/7895)
[SLES-12](http://10.100.12.155/tests/7896)
[TW gnome](http://10.100.12.155/tests/7904#step/firefox/3)
[TW kde](http://10.100.12.155/tests/7907#step/firefox/7)
[Leap gnome](http://10.100.12.155/tests/7905#step/firefox/3)
[Leap kde](http://10.100.12.155/tests/7906#step/firefox/7)
[TW extra_tests](http://10.100.12.155/tests/7928)
[Leap extra_tests](http://10.100.12.155/tests/7926)

### CODE REVIEW

**Comment by okurz** [comment]
The description still references "WIP" and a dummy needle URL. Can you update the description please?

**Comment by dzedro** [comment]
PR is done, all WIP removed

File: lib/x11test.pm [hunk 1]
```diff
@@ -556,56 +554,58 @@ sub start_clean_firefox {
...
-    # store .mozilla configuration as default to avoid popup checks in following tests
-    $self->restart_firefox('cp -rp .mozilla .mozilla_first_run');
+    # restart firefox to trigger default browser pop-up and store .mozilla configuration as default without pop-ups
+    $self->restart_firefox('sync && cp -rp .mozilla .mozilla_first_run', 'opensuse.org');
 }
 
 sub start_firefox_with_profile {
-    my ($self) = @_;
+    my ($self, $url) = @_;
+    $url ||= '/home';
```
Comment (kravciak) [inline lib/x11test.pm]: `||=` vs `//=` - does not matter here, just for info
`||=` - if empty string or the number 0 are valid parameters then they are replaced by default value

File: lib/x11test.pm [hunk 2]
```diff
@@ -641,7 +641,8 @@ sub firefox_check_popups {
 }
 
 sub firefox_open_url {
-    my ($self, $url) = @_;
+    my ($self, $url, $do_not_check_loaded_url) = @_;
+    $do_not_check_loaded_url ||= 0;
```
Comment (kravciak) [inline lib/x11test.pm]: this is default, no?

Comment (dzedro) [reply lib/x11test.pm]: yes

File: lib/x11test.pm [hunk 3]
```diff
@@ -641,7 +641,8 @@ sub firefox_check_popups {
 }
 
 sub firefox_open_url {
-    my ($self, $url) = @_;
+    my ($self, $url, $do_not_check_loaded_url) = @_;
```
Comment (kravciak) [inline lib/x11test.pm]: You call this as `$self->firefox_open_url('http://www.url.com/', do_not_check_loaded_url => 1);` which is 3 parameters. But you parse just 2 params.

Maybe you wanted
```perl
  my ($self, $url, %args) = @_;
  if ($args{do_not_check_loaded_url}) {...}
```

Comment (dzedro) [reply lib/x11test.pm]: why ?

**Pull request merged**